### PR TITLE
csplit: move from getopts to clap

### PR DESCRIPTION
**Pull request by sivachandran**
**with clap**
```bash
csplit 0.0.4
split a file into sections determined by context lines

USAGE:
    [OPTION]... FILE PATTERN...

FLAGS:
    -z, --elide-empty-files    remove empty output files
    -h, --help                 Prints help information
    -k, --keep-files           do not remove output files on errors
        --quiet                same as --silent
    -s, --silent               do not print counts of output file sizes
        --suppress-matched     suppress the lines matching PATTERN
    -V, --version              Prints version information

OPTIONS:
    -n, --digits <DIGITS>           use specified number of digits instead of 2
    -f, --prefix <PREFIX>           use PREFIX instead of 'xx'
    -b, --suffix-format <FORMAT>    use sprintf FORMAT instead of %02d

Output pieces of FILE separated by PATTERN(s) to files 'xx00', 'xx01', ..., and output byte counts of each piece to
standard output.
```
**with getopts**
```bash
$ ./target/debug/coreutils csplit --help

 csplit 0.0.4

 csplit [OPTION]... FILE PATTERN...

 split a file into sections determined by context lines

 Options:
        --help          print usage information
        --version       print name and version number
    -b, --suffix-format FORMAT
                        use sprintf FORMAT instead of %02d
    -f, --prefix PREFIX use PREFIX instead of 'xx'
    -k, --keep-files    do not remove output files on errors
        --suppress-matched
                        suppress the lines matching PATTERN
    -n, --digits DIGITS use specified number of digits instead of 2
    -s, --quiet         do not print counts of output file sizes
    -z, --elide-empty-files
                        remove empty output files


 Reference
Output pieces of FILE separated by PATTERN(s) to files 'xx00', 'xx01', ..., and output byte counts of each piece to standard output.
```

Addresses #1931

### CODE REVIEW

File: src/uu/csplit/src/csplit.rs [hunk 1]
```diff
@@ -18,17 +18,23 @@ mod splitname;
 use crate::csplit_error::CsplitError;
 use crate::splitname::SplitName;
 
+static VERSION: &str = env!("CARGO_PKG_VERSION");
 static SYNTAX: &str = "[OPTION]... FILE PATTERN...";
```
Comment (jhscheer) [inline src/uu/csplit/src/csplit.rs:22]: The name of the application is missing.
The other ports use a function for that like so:
```rust
fn get_usage() -> String {                         
    format!("{} [OPTION]... FILE PATTERN...", executable!())
}                                                  
```

File: src/uu/csplit/src/csplit.rs [hunk 2]
```diff
@@ -704,43 +710,83 @@ mod tests {
...
-        .optopt("f", PREFIX_OPT, "use PREFIX instead of 'xx'", "PREFIX")
-        .optflag("k", KEEP_FILES_OPT, "do not remove output files on errors")
-        .optflag(
-            "",
-            SUPPRESS_MATCHED_OPT,
-            "suppress the lines matching PATTERN",
+        .arg(
+            Arg::with_name(options::QUIET)
+                .long(options::QUIET)
+                .help("same as --silent"),
```
Comment (jhscheer) [inline src/uu/csplit/src/csplit.rs:758]: No need for an extra `arg`, use https://docs.rs/clap/2.33.3/clap/struct.Arg.html#method.visible_alias instead.

File: src/uu/csplit/src/csplit.rs [hunk 3]
```diff
@@ -704,43 +710,83 @@ mod tests {
...
     }
-    if matches.free.len() == 1 {
-        show_error!("missing operand after '{}'", matches.free[0]);
+
+    // get the file to split
+    let file_name = file_value.unwrap();
+
+    let patterns_values = matches.values_of(options::PATTERN);
+    if patterns_values.is_none() {
+        show_error!("missing operand after '{}'", file_name);
```
Comment (jhscheer) [inline src/uu/csplit/src/csplit.rs:783]: clap should take care of these cases if you tell it which `.arg` are `required`, `required_unless`, etc.

File: src/uu/csplit/src/csplit.rs [hunk 4]
```diff
@@ -704,43 +710,83 @@ mod tests {
...
-        .parse(args);
+        .arg(Arg::with_name(options::FILE).hidden(true))
+        .arg(Arg::with_name(options::PATTERN).hidden(true).multiple(true))
+        .after_help(LONG_HELP)
+        .get_matches_from(args);
 
     // check for mandatory arguments
-    if matches.free.is_empty() {
+    let file_value = matches.value_of(options::FILE);
+    if file_value.is_none() {
```
Comment (ycd) [inline src/uu/csplit/src/csplit.rs:773]: I think there is no need for this, afaik clap can handle this

**Review by sylvestre — CHANGES_REQUESTED** [review]
Just like the two other great reviewers did!

**Comment by sivachandran** [comment]
@jhscheer @ycd Thanks for the review. Addressed the comments.

**Comment by sylvestre** [comment]
great, thanks

**Pull request merged**